### ISSUE TITLE
WakaTime detecting VS Code running 24 hrs 

### ISSUE THREAD

**Issue by pychap** [wakatime/vscode-wakatime#163]
Got this message:

> _We noticed some unusual coding activity from your VS Code WakaTime plugin Yesterday:
> You have 24 hrs of coding activity._

> _In rare cases, other VS Code extensions can cause problems with WakaTime. Please make sure your other VS Code extensions are not simulating cursor movements or saving files in the background. This can trick the WakaTime plugin into thinking you were typing even when you weren’t using your VS Code editor._  

  
I recently activated the new VS Code native feature Settings Sync… I wonder if this is the source of the issue??

**Comment by alanhamlett** [comment]
Similar to #145. We'll look into this and release a fix, then update here.

**Comment by nicolasvahidzein** [comment]
![001](https://user-images.githubusercontent.com/29876108/102786468-d3f76900-439f-11eb-80df-529cad404502.JPG)
![002](https://user-images.githubusercontent.com/29876108/102786473-d659c300-439f-11eb-84e7-f530ccadb9a6.JPG)
![003](https://user-images.githubusercontent.com/29876108/102786476-d78af000-439f-11eb-8acc-a0e912de1cb8.JPG)

here are my plugins

**Comment by alanhamlett** [comment]
Fixed in the release today v3.0.4.

**Comment by javero** [comment]
Hi friends, I also have this issue.

Here is the list of my extensions (I do have a lot, sorry 😳  ), most are themes though:
- 4ops.terraform-0.2.1
- abusaidm.html-snippets-0.2.1
- adamcaviness.theme-monokai-dark-soda-0.0.6
- adamcaviness.theme-monokai-dark-soda-1.0.0
- ahmadawais.shades-of-purple-6.12.0
- akamud.vscode-theme-onedark-2.2.3
- akamud.vscode-theme-onelight-2.2.3
- alanwalk.markdown-toc-1.5.6
- alefragnani.bookmarks-12.1.2
- arcticicestudio.nord-visual-studio-code-0.15.0
- atlassian.atlascode-2.8.5
- azemoh.one-monokai-0.5.0
- azemoh.theme-onedark-0.6.0
- be5invis.vscode-icontheme-nomo-dark-1.3.6
- bernardodsanderson.theme-material-neutral-0.9.8
- bibhasdn.unique-lines-1.0.0
- castwide.solargraph-0.21.1
- chef-software.chef-1.9.0
- christian-kohler.npm-intellisense-1.3.1
- christian-kohler.path-intellisense-2.3.0
- codezombiech.gitignore-0.6.0
- davidanson.vscode-markdownlint-0.38.0
- dbaeumer.jshint-0.11.0
- dbaeumer.vscode-eslint-2.1.14
- deerawan.vscode-dash-2.4.0
- dnlytras.nord-wave-0.3.4
- donjayamanne.githistory-0.6.14
- dotjoshjohnson.xml-2.5.1
- dracula-theme.theme-dracula-2.22.3
- dunstontc.viml-0.1.7
- eamodio.gitlens-11.1.3
- editorconfig.editorconfig-0.16.4
- eg2.tslint-1.0.44
- eg2.vscode-npm-script-0.3.13
- emmanuelbeziat.vscode-great-icons-2.1.64
- equinusocio.vsc-community-material-theme-1.4.2
- equinusocio.vsc-material-theme-33.1.2
- equinusocio.vsc-material-theme-icons-1.2.2
- erd0s.terraform-autocomplete-0.0.8
- esbenp.prettier-vscode-5.8.0
- file-icons.file-icons-1.0.28
- formulahendry.auto-close-tag-0.5.10
- formulahendry.code-runner-0.11.2
- foxundermoon.shell-format-7.0.1
- georgewfraser.vscode-javac-0.2.37
- github.vscode-pull-request-github-0.22.0
- golang.go-0.20.2
- groksrc.haml-0.0.2
- hashicorp.terraform-2.4.0
- hashicorp.terraform-2.5.0
- hediet.debug-visualizer-2.0.6
- heptio.jsonnet-0.1.0
- himanoa.python-autopep8-1.0.2
- hoovercj.ruby-linter-1.0.0
- hoovercj.vscode-power-mode-2.2.0
- httpsterio.henna-0.11.0
- idleberg.hopscotch-0.8.1
- ionutvmi.path-autocomplete-1.17.0
- ipedrazas.kubernetes-snippets-0.1.9
- jinsihou.diff-tool-0.0.1
- jmrog.vscode-nuget-package-manager-1.1.6
- johnpapa.winteriscoming-1.4.3
- jolaleye.horizon-theme-vscode-2.0.2
- joshwong.vscode-jmespath-0.0.6
- jprestidge.theme-material-theme-1.0.1
- karyfoundation.theme-karyfoundation-themes-20.0.3
- kirozen.wordcounter-2.3.0
- korekontrol.saltstack-0.0.8
- kuscamara.electron-0.2.5
- kyoh86.sumiiro-0.0.1
- lafe.contextualduplicate-0.2.0
- liviuschera.noctis-10.39.1
- maxfieldwalker.vscode-color-theme-spirited-away-1.3.2 robbowen.synthwave-vscode-0.1.8
- mdickin.markdown-shortcuts-0.12.0
- michelemelluso.code-beautifier-2.3.3
- michelemelluso.gitignore-1.0.1
- mikestead.dotenv-1.0.1
- misogi.ruby-rubocop-0.8.5
- mrmlnc.vscode-duplicate-1.2.1
- ms-azuretools.vscode-docker-1.9.0
- ms-kubernetes-tools.vscode-kubernetes-tools-1.2.1
- ms-mssql.mssql-1.10.0
- ms-python.python-2020.12.424452561
- ms-toolsai.jupyter-2020.12.414227025
- ms-vscode-remote.remote-containers-0.154.2
- ms-vscode.azure-account-0.9.4
- ms-vscode.hexeditor-1.3.0
- ms-vscode.theme-markdownkit-0.1.4
- ms-vscode.theme-materialkit-0.1.4
- narekmal.vscode-run-git-difftool-0.0.5
- okbiem.workspacefolderadder-0.0.4
- paulhoughton.vscode-jscpd-0.3.4
- pkief.material-icon-theme-4.5.0
- qinjia.view-in-browser-0.0.5
- qub.qub-msbuild-0.6.3
- rafaelmaiolla.diff-0.0.1
- rbukovansky.dark-win16-theme-0.0.4
- rebornix.ruby-0.28.1
- redhat.vscode-yaml-0.13.0
- redhat.vscode-yaml-0.14.0
- richie5um2.vscode-sort-json-1.19.2
- rokoroku.vscode-theme-darcula-1.2.3
- rubymaniac.vscode-direnv-0.0.2
- run-at-scale.terraform-doc-snippets-0.4.0
- ryu1kn.partial-diff-1.4.1
- samuelcolvin.jinjahtml-0.15.0
- samuelcolvin.jinjahtml-0.16.0
- sdras.night-owl-2.0.0
- searking.preview-vscode-2.2.5
- secanis.jenkinsfile-support-0.1.0
- shan.code-settings-sync-3.4.3
- shd101wyy.markdown-preview-enhanced-0.5.15
- sldobri.nord-5-stars-1.0.1
- sleistner.vscode-fileutils-3.4.3
- slevesque.vscode-zipexplorer-0.3.1
- solarliner.linux-themes-1.0.1
- streetsidesoftware.code-spell-checker-1.10.2
- sugarpirate.theme-tinacious-contrast-0.0.2
- svipas.prettier-plus-4.2.2
- swyphcosmo.spellchecker-1.3.0
- teabyii.ayu-0.20.1
- tht13.python-0.2.3
- tinkertrain.theme-panda-1.3.0
- travisthieman.better-search-0.0.6
- tyriar.shell-launcher-0.4.1
- vangware.dark-plus-material-2.3.12
- velok.theme-gneak-0.2.1
- visualstudioexptteam.vscodeintellicode-1.2.10
- vscode-icons-team.vscode-icons-11.1.0
- vscoss.vscode-ansible-0.6.0
- waderyan.gitblame-6.0.2
- wakatime.vscode-wakatime-5.0.0
- wayou.vscode-todo-highlight-1.0.4
- wesbos.theme-cobalt2-2.2.5
- whizkydee.material-palenight-theme-2.0.1
- wholroyd.jinja-0.0.8
- wingrunr21.vscode-ruby-0.28.0
- xhi2art.xhitheme-color-0.0.1
- yzhang.markdown-all-in-one-3.4.0
- zhuangtongfa.material-theme-3.9.12
- ziyasal.vscode-open-in-github-1.3.6

I have no idea what could be wrong, last time I installed a new extension was more than a month ago. Perhaps a new version of one of these extensions is messing things up.

Thanks a lot!

**Comment by MPTG94** [comment]
Hi, been having the exact same issue and getting an email about for the past 2 weeks. Don't remember any new extension I installed, just have the VSCode settings sync running, and recently installed the extensions PKief.material-product-icons@1.0.4, but I don't see any reason it would cause an issue

abusaidm.html-snippets@0.2.1
adpyke.vscode-sql-formatter@1.4.4
alefragnani.Bookmarks@13.0.4
alefragnani.project-manager@12.0.1
austin.code-gnu-global@0.2.2
azemoh.one-monokai@0.5.0
brenosolutions.sqlformatter@0.0.7
christian-kohler.npm-intellisense@1.3.1
christian-kohler.path-intellisense@2.3.0
dandric.elasticsearch-snippets@0.1.2
donjayamanne.githistory@0.6.16
donjayamanne.jquerysnippets@0.0.1
eamodio.gitlens@11.3.0
EditorConfig.EditorConfig@0.16.4
esbenp.prettier-vscode@6.3.1
file-icons.file-icons@1.0.29
formulahendry.auto-rename-tag@0.1.6
GitHub.github-vscode-theme@3.0.0
GrapeCity.gc-excelviewer@3.0.41
HookyQR.beautify@1.5.0
IBM.output-colorizer@0.1.2
ilich8086.launcher@0.0.4
jakebathman.mysql-syntax@1.3.1
joelday.docthis@0.7.1
johnpapa.Angular2@11.0.0
liviuschera.noctis@10.39.1
mkaufman.HTMLHint@0.10.0
mohsen1.prettify-json@0.0.3
ms-azuretools.vscode-docker@1.11.0
ms-dotnettools.csharp@1.23.9
ms-python.python@2021.3.680753044
ms-toolsai.jupyter@2021.3.684299474
ms-vscode-remote.remote-wsl@0.54.6
ms-vscode.cpptools@1.2.2
ms-vscode.powershell@2021.2.2
msjsdiag.debugger-for-chrome@4.12.12
naumovs.color-highlight@2.3.0
njpwerner.autodocstring@0.5.4
PKief.material-icon-theme@4.6.0
PKief.material-product-icons@1.0.4
qinjia.seti-icons@0.1.3
redhat.vscode-yaml@0.17.0
shakram02.bash-beautify@0.1.1
spywhere.guides@0.9.3
streetsidesoftware.code-spell-checker@1.10.2
vincaslt.highlight-matching-tag@0.10.0
WakaTime.vscode-wakatime@5.0.1
WallabyJs.quokka-vscode@1.0.353
wcwhitehead.bootstrap-3-snippets@0.1.0
wmaurer.change-case@1.0.0
xabikos.JavaScriptSnippets@1.8.0
yzhang.markdown-all-in-one@3.4.0
zhuangtongfa.material-theme@3.9.15
Zignd.html-css-class-completion@1.20.0

hope this helps.

**Comment by alanhamlett** [comment]
You both have EditorConfig extension installed. Can you try disabling that extension for a day and see if the dashboard no longer shows 24hrs of coding activity?

**Comment by MPTG94** [comment]
@alanhamlett I tried disabling the extension and it no longer counts time when VSCode is minimized. But I had the EditorConfig extension installed for quite a long time, and it has never caused this kind of issue with the WakaTime extension up until recently, so I'm guessing there has been some sort of change that caused this. (Not sure on which side though)

**Comment by MPTG94** [comment]
I tried disabling it and upon some inspection and back and forth between activating and deactivating it didn't seem to make a difference. But since a few days ago, the problem doesn't seem to happen anymore. Might be related to an update made to VSCode itself or something similar. But in any case this no longer happens for me.

**Comment by GaraYaka** [comment]
Just got a mail saying that the usage is over 24 hours. Following is the list of extensions installed in Code;

code --install-extension atlassian.atlascode
code --install-extension DotJoshJohnson.xml
code --install-extension ms-dotnettools.csharp
code --install-extension ms-vscode-remote.remote-wsl
code --install-extension redhat.vscode-yaml
code --install-extension Silvenga.edi-x12-support
code --install-extension WakaTime.vscode-wakatime

**Comment by alanhamlett** [comment]
Fixed with v17.0.4.

**Comment by amsv01** [comment]
Got an email today about unusual activity while I was only coding yesterday. Here is my extension list:

I'm using Wakatime `v24.6.0`.

```
aaronyoung.dark-synthwave-vscode
alefragnani.bookmarks
attilabuti.vscode-mjml
bajdzis.vscode-twig-pack
bmewburn.vscode-intelephense-client
boyswan.glsl-literal
bradlc.vscode-tailwindcss
chrisbibby.hide-node-modules
chrisdias.vscode-opennewinstance
christian-kohler.npm-intellisense
christian-kohler.path-intellisense
circledev.glsl-canvas
cmckni3.dark-plus-mono
codium.codium
dbaeumer.vscode-eslint
deerawan.vscode-faker
devsense.composer-php-vscode
devsense.intelli-php-vscode
devsense.phptools-vscode
devsense.profiler-php-vscode
dracula-theme.theme-dracula
eamodio.gitlens
editorconfig.editorconfig
emmanuelbeziat.vscode-great-icons
enkia.tokyo-night
esbenp.prettier-vscode
formulahendry.auto-close-tag
formulahendry.auto-rename-tag
formulahendry.code-runner
foxundermoon.next-js
geforcelegend.vscode-glsl
github.copilot
github.copilot-chat
github.github-vscode-theme
grapecity.gc-excelviewer
hbenl.vscode-test-explorer
hookyqr.minify
humao.rest-client
jock.svg
johnbillion.vscode-wordpress-hooks
kumar-harsh.graphql-for-vscode
letrieu.expand-region
lokalise.i18n-ally
mblode.twig-language
mblode.twig-language-2
mechatroner.rainbow-csv
mehedidracula.php-namespace-resolver
mkxml.vscode-filesize
mrorz.language-gettext
ms-python.debugpy
ms-python.python
ms-python.vscode-pylance
ms-toolsai.jupyter
ms-toolsai.jupyter-keymap
ms-toolsai.jupyter-renderers
ms-toolsai.vscode-jupyter-cell-tags
ms-toolsai.vscode-jupyter-slideshow
ms-vscode-remote.remote-ssh-edit
ms-vscode-remote.remote-wsl
ms-vscode.live-server
ms-vscode.test-adapter-converter
ms-vsliveshare.vsliveshare
naumovs.color-highlight
neilbrayfield.php-docblocker
netcorext.uuid-generator
nidu.copy-json-path
phproberto.vscode-php-getters-setters
pkief.material-icon-theme
prisma.prisma
redhat.fabric8-analytics
redhat.java
redhat.vscode-xml
redhat.vscode-yaml
rifi2k.format-html-in-php
ritwickdey.liveserver
rowewilsonfrederiskholme.wikitext
samuelcharpentier.remove-non-ascii-chars
sdras.night-owl
shd101wyy.markdown-preview-enhanced
skyapps.fish-vscode
sleepyghost01.vscode-find-in-node-modules
slevesque.shader
streetsidesoftware.code-spell-checker
thenikso.github-plus-theme
thinker.data-size-count
tomashubelbauer.email-viewer
tyriar.lorem-ipsum
vangware.dark-plus-material
viktorqvarfordt.vscode-pitch-black-theme
vincesalvino.dark-plus-black
visualstudioexptteam.intellicode-api-usage-examples
visualstudioexptteam.vscodeintellicode
vmware.vscode-boot-dev-pack
vmware.vscode-spring-boot
vscjava.vscode-java-debug
vscjava.vscode-java-dependency
vscjava.vscode-java-pack
vscjava.vscode-java-test
vscjava.vscode-maven
vscjava.vscode-spring-boot-dashboard
vscjava.vscode-spring-initializr
vscode-icons-team.vscode-icons
vue.volar
weijunyu.vscode-json-path
whatwedo.twig
wmaurer.change-case
wmaurer.vscode-jumpy
wordpresstoolbox.wordpress-toolbox
xdebug.php-debug
xdebug.php-pack
yzhang.markdown-all-in-one
zhuangtongfa.material-theme
zobo.php-intellisense
```
